rag_api/src/deh/guardrail.py

```diff
@@ -0,0 +1,27 @@
+from langchain_core.runnables import chain
+from langchain_core.documents import Document
+from typing import List
+
+from deh import settings
+
+
+class GuardRailException(Exception):
+    pass
+
+
+def similarity_guardrail(threshold):
+    """Guard-rail: Raise exception if no contexts found above threshold."""
+
+    @chain
+    def exception_if_no_results(docs) -> List[Document]:
+        filtered_docs = [
+            doc for doc in docs if doc.metadata["similarity_score"] > threshold
```

Comment:
https://python.langchain.com/api_reference/chroma/vectorstores/langchain_chroma.vectorstores.Chroma.html#langchain_chroma.vectorstores.Chroma.similarity_search_with_score

The score seems to be cosine similarity - lower is more similar / better, right?

```suggestion
            doc for doc in docs if doc.metadata["similarity_score"] <= threshold
```

Reply:
It looks like the default may be a distance function instead of an angle cosine function.  Should we change to the cosine angle function to constrain between -1 and 1?

Reply:
Updated to cosine sim in next PR.